### DISCUSSION TITLE
Spring Web Migration - HttpServletRequest/Response

### DISCUSSION THREAD

**Discussion by frootshoot** [quarkusio/quarkus#41412] [Q&A]
We are looking at migrating a number of Spring applications to Quarkus and a lot of them use HttpServletRequest objects as parameters alongside the standard Header / Path annotations.

As per the documentation we added "quarkus-undertow" to the dependencies and ran the application. On compilation the "GET" methods had warnings about using body parameters, and when calling any of the endpoints we got a 415 error in the logs.

Is there anything we are missing in order to port these applications without having to change them to use other context parameters?

In addition a small number of services were using  HttpServletResponse to send a file back as an attachment - the docs imply this is supported as the object is, but again no joy ?

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
Thanks for reporting this!

Are you using `quarkus-rest` or `quarkus-resteasy`?

**Reply by frootshoot** [reply]
The dependencies we were using were: "quarkus-rest-jackson", "quarkus-spring-web", "quarkus-undertow"

**Reply by geoand** [reply]
Can you try `quarkus-resteasy-jackson` instead of `quarkus-rest-jackson`?

**Reply by frootshoot** [reply]
We have tried a few combinations.

First the good news - when we simply switch "quarkus-rest-jackson" to "quarkus-resteasy-jackson", the server works as expected with an HttpServletRequest. (not tested HttpServletResponse in this case yet).

However, in a server where we also have a dependency on "quarkus-rest-client-jackson" - this initially fails with a DeploymentException "Mixing Quarkus REST and RESTEasy Classic server parts is not supported"

If we change that to "quarkus-resteasy-client-jackson" as well - we now get an exception when we run a test case where we are trying to inject the Rest Client interface (with a @RegisterRestClient annotation) - "Invalid use of io.quarkus.test.InjectMock - the injected bean does not declare a CDI normal scope but: jakarta.enterprise.context.Dependent" 

I hope this helps :)

**Reply by geoand** [reply]
When using `quarkus-resteasy-client-jackson`, you can also configure the scope of the REST Clientm, see https://quarkus.io/guides/resteasy-client#create-the-configuration for an example.

That should get you over the exception you are seeing in the mock test.

Note, that I am trying to giving you information about the path of least resistance for your migration. There are certainly more things that can be done that would involve some rewrites.

Let me know how it goes please.
